site/src/components/AuditLogRow/AuditLogRow.tsx

```diff
@@ -0,0 +1,217 @@
+  organization: "organization",
+  template: "template",
+  template_version: "template version",
+  user: "user",
+  workspace: "workspace",
+}
+
+const readableActionMessage = (auditLog: AuditLog) => {
+  return `${actionLabelByAction[auditLog.action]} ${
+    resourceLabelByResourceType[auditLog.resource_type]
+  }`
+}
+
+export interface AuditLogRowProps {
+  auditLog: AuditLog
+  // Useful for Storybook
+  defaultIsDiffOpen?: boolean
+}
+
+export const AuditLogRow: React.FC<AuditLogRowProps> = ({
+  auditLog,
+  defaultIsDiffOpen = false,
+}) => {
+  const styles = useStyles()
+  const [isDiffOpen, setIsDiffOpen] = useState(defaultIsDiffOpen)
+  const diffs = Object.entries(auditLog.diff)
+  const shouldDisplayDiff = diffs.length > 0
+
+  const toggle = () => {
+    if (shouldDisplayDiff) {
+      setIsDiffOpen((v) => !v)
+    }
+  }
+
+  return (
+    <TableRow
+      key={auditLog.id}
+      hover={shouldDisplayDiff}
+      data-testid={`audit-log-row-${auditLog.id}`}
+    >
+      <TableCell className={styles.auditLogCell}>
+        <Stack
+          style={{ cursor: shouldDisplayDiff ? "pointer" : undefined }}
+          direction="row"
+          alignItems="center"
+          className={styles.auditLogRow}
+          tabIndex={0}
+          onClick={toggle}
+          onKeyDown={(event) => {
+            if (event.key === "Enter") {
+              toggle()
+            }
+          }}
+        >
+          <Stack
+            direction="row"
+            alignItems="center"
+            justifyContent="space-between"
+            className={styles.auditLogRowInfo}
+          >
+            <Stack direction="row" alignItems="center">
+              <UserAvatar username={auditLog.user?.username ?? ""} />
+              <div>
+                <span className={styles.auditLogResume}>
+                  <strong>{auditLog.user?.username}</strong> {readableActionMessage(auditLog)}{" "}
+                  <ResourceLink
+                    resource={auditLog.resource}
+                    resourceType={auditLog.resource_type}
+                  />
+                </span>
+                <span className={styles.auditLogTime}>{createDayString(auditLog.time)}</span>
+              </div>
+            </Stack>
+
+            <Stack direction="column" alignItems="flex-end" spacing={1}>
+              <Pill
+                type={pillTypeByHttpStatus(auditLog.status_code)}
+                text={auditLog.status_code.toString()}
+              />
+              <Stack direction="row" alignItems="center" className={styles.auditLogExtraInfo}>
+                <div>
+                  <strong>IP</strong> {auditLog.ip}
+                </div>
+                <div>
+                  <strong>Agent</strong> {auditLog.user_agent}
+                </div>
+              </Stack>
+            </Stack>
+          </Stack>
+
+          <div className={shouldDisplayDiff ? undefined : styles.disabledDropdownIcon}>
```

Comment:
I wonder if it's better to omit the dropdown icon when there's no diff? Also just fyi, I made an Expander component recently that labels the arrows "More" and "Less". It might be too wordy for the audit log though!

Reply:
> I wonder if it's better to omit the dropdown icon when there's no diff? 

I like to keep it and just disable it to avoid "layout shift"/different layout from a component that is expandable and another that is not, but no strong opinion.

> Also just fyi, I made an Expander component recently that labels the arrows "More" and "Less". It might be too wordy for the audit log though!

Nice! I think they handle diff cases. Here, the row should be clickable. The icon is more of a "hint" for the user that they can click or interact with the element.